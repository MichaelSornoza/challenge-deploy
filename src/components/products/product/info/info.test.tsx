import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Info from '.'

const mockProps = {
  total: 100,
  price: 25,
  amount: 4
}

describe('Info component', () => {
  it('renders the price correctly', () => {
    render(<Info {...mockProps} />)
    const priceElement = screen.getByText(/\$\s*25/i)
    expect(priceElement).toBeInTheDocument()
  })

  it('renders the amount correctly', () => {
    render(<Info {...mockProps} />)
    const amountElement = screen.getByText(`x ${mockProps.amount}`)
    expect(amountElement).toBeInTheDocument()
  })

  it('renders the total correctly', () => {
    render(<Info {...mockProps} />)
    const totalElement = screen.getByText(/\s*\$\s*100/i)
    expect(totalElement).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { container } = render(<Info {...mockProps} />)
    expect(container).toMatchSnapshot()
  })
})
