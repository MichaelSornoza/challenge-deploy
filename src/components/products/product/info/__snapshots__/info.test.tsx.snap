// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component matches snapshot 1`] = `
<div>
  <footer
    class="cart-product-footer"
  >
    <div
      class="cart-price"
    >
      <div>
        Precio: 
        <span>
          $ 
          25
        </span>
      </div>
      <div>
        <span
          class="cart-counter"
        >
          x 
          4
        </span>
      </div>
    </div>
    <div
      class="total"
    >
      Total: 
      <span>
        $ 
        100
      </span>
    </div>
  </footer>
</div>
`;
